camera_alignment_core/alignment_utils/alignment_qc.py

```diff
@@ -0,0 +1,363 @@
+import logging
+from typing import Dict, Optional, Tuple
+
+import numpy as np
+from numpy.typing import NDArray
+from scipy.spatial import distance
+from skimage import metrics
+from skimage import transform as tf
+
+from ..constants import LOGGER_NAME
+from .get_center_z import get_center_z
+
+log = logging.getLogger(LOGGER_NAME)
+
+
+class AlignmentQC:
+    def __init__(
+        self,
+        reference: NDArray[np.uint16] = None,
+        moving: NDArray[np.uint16] = None,
+        reference_seg: NDArray[np.uint16] = None,
+        moving_seg: NDArray[np.uint16] = None,
+        ref_mov_coor_dict: Dict[Tuple[int, int], Tuple[int, int]] = None,
+        rev_coor_dict: Dict[Tuple[int, int], Tuple[int, int]] = None,
+        tform: tf.SimilarityTransform = None,
+    ):
+        self.reference = reference
+        self.moving_source = moving
+
+        if reference is not None:
+            self.ref_origin = get_center_z(reference)
+        else:
+            self.ref_origin = 0
+        if moving is not None:
+            self.mov_origin = get_center_z(moving)
+        else:
+            self.mov_origin = 0
+
+        self.reference_seg = reference_seg
+        self.moving_seg = moving_seg
+
+        self.ref_mov_coor_dict = ref_mov_coor_dict
+        self.rev_coor_dict = rev_coor_dict
+
+        if moving is not None and tform is not None:
```

Comment:
can moving be null here?  Not that a safety check is a bad thing, and tform clearly could be.